docs/manual/src/udl/custom_types.md

```diff
@@ -1,72 +1,110 @@
 # Custom types
 
-Custom types allow you to extend the UniFFI type system to support types from your Rust crate or 3rd
-party libraries. This works by converting to and from some other UniFFI type to move data across the
-FFI. You must provide a `UniffiCustomTypeConverter` implementation to convert the types.
+Custom types offer a way to use types in your interface that don't derive one of the UniFFI type traits (`uniffi::Record`, `uniffi::Enum`, `uniffi::Object`, etc.).
+Instead, custom types are converted to/from a type that does derive those traits, called a "bridge type", when being passed across the FFI.
+
+Custom types are often used with structs that wraps a builtin type, for example `Guid(String)`.
+These types can be passed across the FFI as the builtin type, which can be more efficient than passing a struct.
+The foreign bindings will treat these types as the builtin type, for example `Guid` could appear a string to the foreign code.
+
+Custom types can also be customized on the foreign side.  For example, a URL could be:
+* Represented by the `url::Url` type in Rust
+* Passed across the FFI as a string
+* Represented by the `java.net.URL` type in Kotlin
 
 ## Custom types in the scaffolding code
 
-Consider the following trivial Rust abstraction for a "handle" which wraps an integer:
+### custom_type!
+
+Use the `custom_type!` macro to define a new custom type.
 
 ```rust
-pub struct Handle(i64);
+
+// Some complex struct that can be serialized/deserialized to a string.
+use some_mod::SerializableStruct;
+
+// When passing `SerializableStruct` objects to the foreign side, they will
+// be converted to a `String`, then `String` will be used as the bridge type
+// to pass the value across the FFI. Conversly, when objects are passed to
+// Rust, they will be passed across the FFI as a String then converted to
+// `SerializableStruct`.
+//
+// The `Into` / `TryInto` traits will be used for the conversion.
+uniffi::custom_type!(SerializableStruct, String);
 ```
 
-In this trivial example, the simplest way to expose this is with a macro.
+Default conversions to the bridge type:
+
+- Values passed to the foreign code will be converted using `Into<String>` then lowered as a `String` value.
+- Values passed to the Rust code will lifted as a `String` then converted using `<String as TryInto<SerializableStruct>>`.
+- The `TryInto::Error` type can be anything that implements `Into<anyhow::Error>`.
+- `TryFrom<String>` and `From<SerializableStruct>` will also work, using the blanket impl from the core library. 
+
+### custom_type! with manual conversions
 
+You can also manually specify the conversions by passing extra params to the macro:
```

Comment:
like above, maybe add to the end here ", for cases where the trait implementations do not already exist, or they aren't desirable in this context for some reason"?